src/dotnet/commands/dotnet-build/IncrementalCache.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+
+namespace Microsoft.DotNet.Tools.Build
+{
+    internal class IncrementalCache
+    {
+        private const string InputsKeyName = "inputs";
+        private const string OutputsKeyNane = "outputs";
+
+        public CompilerIO CompilerIO { get; }
+
+        public IncrementalCache(CompilerIO compilerIO)
+        {
+            CompilerIO = compilerIO;
+        }
+
+        public static void WriteToFile(string cacheFile, IncrementalCache cache)
```

Comment:
Should this just be a instance method?  Then you can just call it cache.WriteToFile(fileName), instead of IncrementalCache.WriteToFile(fileName, cache)